### PR TITLE
infra: added github configuration file for dependabot with commit prefix

### PR DESCRIPTION
**Pull request by nishant-parekh**

### CODE REVIEW

**Comment by nishant-parekh** [comment]
> hey, this is very nice indeed! but I'd prefer if you add a config file https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#commit-message and set the prefix to `depbot`

I made the changes per your suggestions..

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (yuvalk) [inline .github/dependabot.yml:11]: why not daily?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    commit-message:
+      prefix: "depbot"
+  - package-ecosystem: "pip"
+      directory: "/ztp/resource-generator/tools/cachito"
+      schedule:
+        interval: "weekly"
```
Comment (yuvalk) [inline .github/dependabot.yml:17]: why not daily?

Comment (nishant-parekh) [reply .github/dependabot.yml]: Thought there might be too many PRs but a daily check on keeping dependencies up to date seems like a good idea none the less.. updated..

File: .github/dependabot.yml (hunk 1 above)
Comment (nishant-parekh) [reply .github/dependabot.yml]: updated

**Comment by yuvalk** [comment]
can you plz make commit msg explain the why rather than the what?

**Comment by SchSeba** [comment]
/override ci/prow/e2e-gcp-ovn

nice work!

**Pull request merged**